src/components/input/IconButton.js

```diff
@@ -0,0 +1,79 @@
+import classnames from 'classnames';
+
+import { downcastRef } from '../../util/typing';
+
+import ButtonBase from './ButtonBase';
+
+/**
+ *
+ * @typedef {import('../../types').IconComponent} IconComponent
+ * @typedef {import('../../types').PresentationalProps} CommonProps
+ * @typedef {import('./ButtonBase').ButtonCommonProps} ButtonCommonProps
+ * @typedef {import('./ButtonBase').HTMLButtonAttributes} HTMLButtonAttributes
+ *
+ * @typedef IconButtonProps
+ * @prop {IconComponent} [Icon] - reference to an icon function component
+ *   to render in this button, e.g. CautionIcon
+ * @prop {'sm'|'md'|'lg'} [size='md']
+ * @prop {boolean} [responsive=true] - Apply dimensions for touch
+ *   interfaces to ensure touch target is large enough.
+ * @prop {'primary'|'secondary'|'dark'} [variant='secondary']
+ * @prop {string} title - Required for `IconButton` as there is no text label
+ */
+
+/**
+ * Render a button that only contains an icon.
+ *
+ * @param {CommonProps & ButtonCommonProps & IconButtonProps & HTMLButtonAttributes} props
+ */
+export default function IconButton({
+  children,
+  classes,
+  elementRef,
+
+  pressed,
+  expanded,
+
+  Icon,
+  responsive = true,
```

Comment:
I realize that I should rename this prop to reflect its `true` value (and default to `false`) to match props conventions we're establishing.

Reply:
(Sorted).